### DISCUSSION TITLE
questions relating to segments

### DISCUSSION THREAD

**Discussion by wfro64** [opencrx/opencrx#81] [Q&A]
Hi,

Firstly thanks for setting up this discussion area.

I have a few questions relating to segments please:

1. I read somewhere in the documentation that it is possible to move a segment from one server to another server. How can this be done?
2. Is there a query to delete an entire segment?
3. Is there a query to change the segment name?

Thanks,
Franz

_Originally posted by @fschon in https://github.com/opencrx/opencrx-custom/discussions/4#discussioncomment-320772_

**Comment by nyemiks** [comment]
This is are nice questions. I am also interested in knowing the answers. 

I have attempted calling ref delete on a segment.  It did not work.

**Comment by wfro64** [comment]
A utility for managing segments would no doubt be very useful. This utility (e.g. DbSegmentUtils), would generate the SQL code in the form of a SQL script for the following use cases:
* Delete segment (DELETE FROM T WHERE ...)
* Rename segment (UPDATE T WHERE ...)
* Extract segment (INSERT INTO T SELECT ...)

This would also allow the call from a (bash) script and the generated SQL code could be post-processed if necessary.

**Reply by fschon** [reply]
Thanks for looking at this.
Would it be possible to generate these kind of scripts via OpenMdx when generating a project? So if any customization were to be made these would automatically be generated. auto-generation also makes them more reliable.
Also important would be to delete associated users, subjects and principals.

**Reply by wfro64** [reply]
This tool would be based on the same (generated and model-derived) database schema used by DbSchemaUtils.

**Reply by nyemiks** [reply]
Wow ! This will be a very useful feature.

**Reply by wfro64** [reply]
See #82

**Comment by wfro64** [comment]
See https://github.com/opencrx/opencrx/issues/82
